website/docs/reference/node-selection/methods.md

```diff
@@ -83,26 +83,34 @@ $ dbt test --models test_name:range_min_max     # run all instances of a custom
 ```
 
 ### The "state" method
-<Changelog>New in v0.18.0</Changelog>
+<Changelog>
+    - **v0.18.0** introduced `state:new` and `state:modified`
+    - **v0.21.0** introduced `modified` sub-selectors, and handling for upstream macro dependencies
+</Changelog>
 
 **N.B.** State-based selection is a powerful, complex feature. Read about [known caveats and limitations](node-selection/state-comparison-caveats) to state comparison.
 
 The `state` method is used to select nodes by comparing them against a previous version of the same project, which is represented by a [manifest](artifacts/manifest-json). The file path of the comparison manifest _must_ be specified via the `--state` flag or `DBT_ARTIFACT_STATE_PATH` environment variable.
 
 `state:new`: There is no node with the same `unique_id` in the comparison manifest
 
-`state:modified`: Everything new, plus any changes to:
-* file/node contents
-* configs (`materialized`, `bind`, `transient`, `quote`, etc.)
-* descriptions (top-level and/or column-level, depending on `persist_docs`)
-* database representations (user-input `database`, `schema`, `alias`, irrespective of `target` + `generate_x_name` macros)
+`state:modified`: All new nodes, plus any changes to existing nodes.
 
 ```bash
 $ dbt test --models state:new            # run all tests on new models + and new tests on old models
 $ dbt run --models state:modified        # run all models that have been modified
-$ dbt ls --models state:modified     # list all modified nodes (not just models)
+$ dbt ls --models state:modified         # list all modified nodes (not just models)
 ```
 
+Because state comparison is complex, and everyone's project is different, dbt supports subselectors that include a subset of the full `modified` criteria:
+- `state:modified.body`: Changes to node body (e.g. model SQL, seed values)
+- `state:modified.configs`: Changes to any node configs, excluding `database`/`schema`/`alias`
+- `state:modified.relation`: Changes to `database`/`schema`/`alias` (the database representation of this node), irrespective of `target` values or `generate_x_name` macros
+- `state:modified.persisted_descriptions`: Changes to relation- or column-level `description`, _if and only if_ `persist_docs` is enabled at each level
+- `state:modified.macros`: Changes to upstream macros (whether called directly or indirectly by another macro)
+
+Remember that `state:modified` includes _all_ of the criteria above, as well as some extra resource-specific criteria: a source's `freshness` property has changed, an exposure's `maturity` has changed, etc.
```

Comment:
Good nit! Honestly, I think this logic is overdue for some clarification _and_ documentation. The general thrust is, if you change something, it should probably be a change, innit? For now, I'm going to link to the relevant methods in the source code (ugh), because they're pretty well enumerated there.